backend/graphql/resolvers/courseResolvers.ts

```diff
@@ -62,22 +124,32 @@ const courseResolvers = {
       const role = await authService.getUserRoleByAccessToken(accessToken);
       const attributes = getCourseVisibilityAttributes(role);
 
-      return courseService.getCourses(attributes);
+      return courseService
+        .getCourses(attributes)
+        .then((courses) => courses.map(serializeCourseResponse));
     },
   },
   Mutation: {
     createCourse: async (
       _parent: undefined,
-      { course }: { course: CreateCourseRequestDTO },
-    ): Promise<CourseResponseDTO> => {
-      const newCourse = await courseService.createCourse(course);
+      { course }: { course: SerializedCreateCourseRequestDTO },
+    ): Promise<SerializedCourseResponseDTO> => {
+      const newCourse = await courseService
+        .createCourse(deserializeCourseRequest(course))
+        .then(serializeCourseResponse);
       return newCourse;
     },
     updateCourse: async (
       _parent: undefined,
-      { id, course }: { id: string; course: UpdateCourseRequestDTO },
-    ): Promise<CourseResponseDTO | null> => {
-      return courseService.updateCourse(id, course);
+      { id, course }: { id: string; course: SerializedUpdateCourseRequestDTO },
+    ): Promise<SerializedCourseResponseDTO | null> => {
+      return courseService
+        .updateCourse(id, deserializeCourseRequest(course))
+        .then((serializedCourse) => {
+          if (serializedCourse)
+            return serializeCourseResponse(serializedCourse);
+          return null;
+        });
```

Comment:
In terms of naming, are we expecting a `serializedCourse` response? I don't think so, because we would serialize it when it's returned -- think the param name on ln#148 doesn't make sense then

nitpick syntax again, does this work?

```suggestion
        .then((serializedCourse) => serializedCourse ? seralizeCourseResponse(serailizedCourse) : null;
```

Reply:
Yeah good catch, I just messed up the naming